### PR TITLE
Show Antibiotics in grid format and auto populate antibiotics on species select

### PR DESCRIPTION
**Pull request by 9sneha-n**
### :pushpin: References

-   **Issue:** Closes #?https://app.clickup.com/t/8694ap44d
- https://app.clickup.com/t/8694ap48n
- 8694ap44d
- 8694ap48n

### :memo: Implementation
1. On species change, pre populate all antibiotic options in grid
2. handle AST Result or Value data entry
3. On reset of species, reset antibiotic details.

### :video_camera: Screenshots/Screen capture

https://github.com/EyeSeeTea/amr-surveys/assets/83749675/5d50d20c-19b5-4863-a5c6-d4bdc5b762a9

### CODE REVIEW

File: src/webapp/components/survey/hook/useGridSection.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                return antiBioticSet;
+            })
+        )
+            .compact()
+            .value();
+
+        setAntibioticSets(antibioticGroups);
+    }, [antibioticStage.sections, getAntibioticOptions, gridOptions, speciesSection.questions]);
+
+    const updateSpeciesQuestion = useCallback(
```
Comment (anagperal) [inline src/webapp/components/survey/hook/useGridSection.ts:70]: I don't know the context but, this is ok? `if isSpeciesQuestion` is going to updateQuestion everytime `if`s inside are fulfilled and also at the end after the `if isSpeciesQuestion`, the question is going to be updated again just with question? Maybe it's better to create a question object to be updated and at the end pass this object to the function updateQuestion?

**Review by anagperal — CHANGES_REQUESTED** [review]
thanks @9sneha-n I have added just one comment that it's more a question that a request of changes because I don't know the context of the task. Also nice the creation of components + hooks :tada:

File: src/webapp/components/survey/hook/useGridSection.ts (hunk 1 above)
Comment (9sneha-n) [reply src/webapp/components/survey/hook/useGridSection.ts:70]: Hi @anagperal ,yes this is correct. Whenever a species question is updated, we need to reset all the associated antibiotic questions . So, the last updateQuestion updates the species question itself and the rest of the updatequestion calls are for the antiobiotic questions which are of 4 types :)

**Comment by 9sneha-n** [comment]
> thanks @9sneha-n I have added just one comment that it's more a question that a request of changes because I don't know the context of the task. Also nice the creation of components + hooks 🎉 (code-only review)
thanks @anagperal

**Review by anagperal — APPROVED** [review]
thanks @9sneha-n !

**Pull request merged**